corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -136,6 +138,8 @@ def _process_main_forms(self):
             while True:
                 new_wrapped_form = self.queues.get_next()
                 if not new_wrapped_form:
+                    if self.queues.full:
+                        sleep(0.01)  # swap greenlets
```

Comment:
this seems like a strange place for this check. surely this only applies if `new_wrapped_form` is not falsy?

They way I'm reading the intention is that this is meant to slow down the growth of the queue when it's full. In that case I would put the check in an else condition after line 133:
```python
if self.queues.try_obj(case_ids, wrapped_form):
    pool.spawn(self._migrate_form_and_associated_models_async, wrapped_form)
elif self.queues.full:
    sleep(0.01)
    
```

Reply:
If there isn't a `new_wrapped_form`, it means that the queues are either empty or everything is locked. If it's the latter, and the queue is also "full", I still think this would be a reasonable spot to make sure something was released.

I guess just slowing the growth rather than killing it could be good though.

Reply:
ok, I had forgotten about the second case of everything being locked. This still won't 'block' since it will break still break out the loop.

Reply:
updated it anyway. I like the other spot as a way to slow it rather than kill it ceaad42